src/VisualStudio/Xaml/Impl/Implementation/XamlTextViewCreationListener.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.CodeAnalysis.Xaml.Diagnostics.Analyzers;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml
+{
+    [Export(typeof(IVsTextViewCreationListener))]
+    [ContentType(ContentTypeNames.XamlContentType)]
+    [TextViewRole(PredefinedTextViewRoles.PrimaryDocument)]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    internal sealed partial class XamlTextViewCreationListener : IVsTextViewCreationListener
+    {
+        private readonly System.IServiceProvider _serviceProvider;
+        private readonly VisualStudioWorkspaceImpl _vsWorkspace;
+        private readonly ICommandHandlerServiceFactory _commandHandlerService;
+        private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactory;
+        private readonly Lazy<RunningDocumentTable> _rdt;
+        private readonly IVsSolution _vsSolution;
+        private uint? _solutionEventsCookie;
+        private uint? _rdtEventsCookie;
+
+        internal ICommandHandlerServiceFactory CommandHandlerServiceFactory
+        {
+            get
+            {
+                return _commandHandlerService;
+            }
+        }
+
+        [ImportingConstructor]
+        public XamlTextViewCreationListener(
+            [Import(typeof(SVsServiceProvider))] System.IServiceProvider services,
+            ICommandHandlerServiceFactory commandHandlerServiceFactory,
+            IVsEditorAdaptersFactoryService editorAdaptersFactoryService,
+            IXamlDocumentAnalyzerService analyzerService,
+            VisualStudioWorkspaceImpl vsWorkspace)
+        {
+            _serviceProvider = services;
+            _commandHandlerService = commandHandlerServiceFactory;
+            _editorAdaptersFactory = editorAdaptersFactoryService;
+            _vsWorkspace = vsWorkspace;
+            _rdt = new Lazy<RunningDocumentTable>(() => new RunningDocumentTable(_serviceProvider));
+            _vsSolution = (IVsSolution)_serviceProvider.GetService(typeof(SVsSolution));
+
+            AnalyzerService = analyzerService;
```

Comment:
I guess since you're still getting a Document handed to you in the analyzer, you're still fine. Have your analyzer service inherit from ILanguageService, and put an [ExportLanguageService] attribute on it. Then in your analyzer you can call document.Project.Services.GetRequiredService(), which does jump back to MEF.